b2dEditor.py

```diff
@@ -0,0 +1,347 @@
+        pos_root_list = []
+        for j in shapes[i]:
+
+            pos_dict = {}
+
+            pos_dict["x"] = j.pos[0] / edit_area.rect[2]
+            pos_dict["y"] = j.pos[1] / edit_area.rect[3]
+
+            pos_root_list.append(pos_dict)
+
+        polygons.append([pos_root_list])
+
+    origin_pos = {}
+    origin_pos["x"] = origin.pos[0] / edit_area.rect[2]
+    origin_pos["y"] = origin.pos[1] / edit_area.rect[3]
+    rigid_body["origin"] = origin_pos
+    rigid_body["polygons"] = polygons
+    rigid_body["circles"] = []
+    rigid_body["shapes"] = []
+    json_file["rigidBody"] = rigid_body
+
+    if __name__ == "__main__":
+        with open(export_file_name, "w") as export_file:
+            json.dump(json_file, export_file, indent=2, sort_keys=True)
+    else:
+        return_json = json_file
+        quit = True
+
+def load_node_json():
+    import_file_name = filedialog.askopenfilename(filetypes=[("JSON files", "*.json")], initialdir="~")
+    if import_file_name == "" or import_file_name == ():
+        return
+
+    with open(import_file_name, "r") as import_file:
+        json_file = json.load(import_file)
+
+
+    for i in json_file["rigidBody"]["polygons"]:
+        for j in i:
+            for k in j:
+                x = int(k["x"] * edit_area.rect[2])
+                y = int(k["y"] * edit_area.rect[3])
+                add_node((x, y))
+        add_shape()
+
+    origin.set_pos((int(json_file["rigidBody"]["origin"]["x"] * edit_area.rect[2]), int(json_file["rigidBody"]["origin"]["y"] * edit_area.rect[3])))
+
+button_flag = False
+
+class Button():
+    def __init__(self, size, pos, command, text):
+        buttons.append(self)
+        self.rect = pygame.Rect(pos, size)
+        self.command = command
+        self.text = text
+        self.norm_color = (50, 50, 50)
+        self.color = self.norm_color
+        self.hover_color = (100, 100, 100)
+
+    def set_color(self, color):
+        self.color = color
+
+    def handle_mouse(self, event):
+        global button_flag
+
+        if self.rect.collidepoint(pygame.mouse.get_pos()):
+            self.set_color(self.hover_color)
+        else:
+            self.set_color(self.norm_color)
+
+        if event.type == pygame.MOUSEBUTTONDOWN:
+            if event.button == 1:
+                if self.rect.collidepoint(event.pos):
+                    if button_flag == False:
+                        exec(self.command)
+                        button_flag = True
+
+        if event.type == pygame.MOUSEBUTTONUP:
+            if event.button == 1:
+                button_flag = False
+
+
+# Define Node class
+class Node():
+    def __init__(self):
+        self.norm_color = (0, 0, 200)
+        self.select_color = (0, 100, 0)
+        self.color = self.norm_color
+        self.pos = (0, 0)
+        self.rect = None
+
+    def set_pos(self, pos):
+        self.pos = pos
+
+    def set_color(self, color):
+        self.color = color
+
+class OriginNode():
```

Comment:
Class definitions do not need parentheses (`()`).

Reply:
Also, two blank lines are usually expected before class/function declaration

Reply:
What is the purpose of having separate `Node` and `OriginNode` classes? They are both almost exactly the same except for two fields and the latter could be replaced by the former